lib/Dialect/Dialect.cpp

```diff
@@ -281,7 +281,8 @@ bool llvm_dialects::detail::isOperationDecl(llvm::StringRef fn,
     if (!fn.starts_with(mnemonic))
       return false;
 
-    return fn[mnemonic.size()] == '.';
+    return fn.size() + 2 >= mnemonic.size() && fn[mnemonic.size()] == '_' &&
```

Comment:
Can't you just use `.substr()`?

Reply:
The "__" will not be at the end of the intrinsic name. And substr would make a copy and allocate

Reply:
Hm, yeah, I missed that this will not be at the end of the intrinsic name.